File: cardbattlemodule/src/main/java/com/syahputrareno975/cardbattlemodule/interfaces/ShopStreamEventController.java
package com.syahputrareno975.cardbattlemodule.interfaces;

import com.syahputrareno975.cardbattlemodule.model.card.CardModel;
import com.syahputrareno975.cardbattlemodule.model.player.PlayerModel;

public interface ShopStreamEventController {
    void getMyPlayerData(String id);
    void getAllCardInShop(PlayerModel p);
    void buyCardFromShop(PlayerModel p,CardModel c);
    void sellCardToShop(PlayerModel p,CardModel c);
    void upgradeCard(PlayerModel p,CardModel c);
    void addDeckSlot(PlayerModel p,int typeSlot);
    void leaveShop(SimpleUnit action);
}

File: app/src/main/java/com/syahputrareno975/simplecardbattle/interfaces/ShopStreamEventController.kt
package com.syahputrareno975.simplecardbattle.interfaces

import com.syahputrareno975.simplecardbattle.model.card.CardModel
import com.syahputrareno975.simplecardbattle.model.player.PlayerModel

interface ShopStreamEventController {
    fun getMyPlayerData(id : String)
    fun getAllCardInShop(p : PlayerModel)
    fun buyCardFromShop(p: PlayerModel,c : CardModel)
    fun sellCardToShop(p: PlayerModel,c : CardModel)
    fun addDeckSlot(p :PlayerModel,typeSlot : Int)
    fun leaveShop(action : () -> Unit)
}